rules/xcodeproj.bzl

```diff
@@ -793,6 +801,9 @@ https://www.rubydoc.info/github/CocoaPods/Xcodeproj/Xcodeproj/Constants
         "installer": attr.label(executable = True, default = Label("//tools/xcodeproj_shims:installer"), cfg = "host"),
         "build_wrapper": attr.label(executable = True, default = Label("//tools/xcodeproj_shims:build-wrapper"), cfg = "host"),
         "additional_files": attr.label_list(allow_files = True, allow_empty = True, default = [], mandatory = False),
+        "custom_build_options_by_config": attr.string_list_dict(allow_empty = True, default = {}, mandatory = False, doc = """\
+The custom build options for different build configs. The key must be `Debug` or `Release`. The value is a list of Bazel build options.
```

Comment:
I was wondering if there are any other possible value... but checked xcodegen and they also only explicitly support release and debug 👍

Reply:
We plan to support release variants, in which case each key will correspond to a release variant.